File: 04/04.py

# coding: utf-8

# In[1]:

from hashlib import md5


# In[2]:

graphics = False


# In[3]:

def main():
#    with open("input.txt") as file:
#        data = file.read().strip()
#    if "\n" in data:
#        data = data.split("\n")
#    data = process_input(data)
    data = "yzbqklnj"
    part1 = evalp1(data, graphics)
    print("Part 1:")
    print(part1[0])
    print("Part 2:")
    print(evalp2(data, part1[1], graphics))


# In[ ]:

def process_input(inp):
    pass


# In[4]:

def evalp1(data, graphics):
    i = 0
    while not md5(bytes(data + str(i), encoding="utf-8")).hexdigest().startswith("00000"):
        i += 1
    return (i, None)


# In[5]:

def evalp2(data, part1, graphics):
    i = 0
    while not md5(bytes(data + str(i), encoding="utf-8")).hexdigest().startswith("000000"):
        i += 1
    return i


# In[6]:

main()

File: 04/04.rb

require 'digest'

GRAPHICS = false

def main
#  input = File.read("input.txt").chomp
#  input = input.split("\n") if input.include?("\n")
#  data = process_input(input)
  data = "yzbqklnj"
  part1 = evalp1(data, GRAPHICS)
  puts("Part 1:", part1[0])
  puts("Part 2:", evalp2(data, part1[1], GRAPHICS))
end

def process_input(input)
  
end

def evalp1(data, graphics)
  i = 0
  until Digest::MD5.hexdigest(data + i.to_s).start_with?("00000") do i += 1 end
  return [i, nil]
end

def evalp2(data, part1, graphics)
  i = 0
  until Digest::MD5.hexdigest(data + i.to_s).start_with?("000000") do i += 1 end
  return i
end

main
